src/notify.test.ts

```diff
@@ -1,23 +1,38 @@
-import * as electron from 'electron'; // tslint:disable-line no-implicit-dependencies
-import settings from 'electron-settings';
+const mockGetSettings = jest.fn();
+const mockOn = jest.fn();
+const mockShow = jest.fn();
+const inMockConstructor = jest.fn();
+class Notification {
+  public on = mockOn;
+  public show = mockShow;
+  constructor() {
+    inMockConstructor();
+  }
+}
+/* doMock instead of mock to prevent hoisting above class and const declarations*/
+jest.doMock('electron', () => ({
+  Notification
+}));
+jest.doMock('electron-settings', () => ({
+  get: mockGetSettings
+}));
```

Comment:
What do you think of this approach? This test no longer uses a special mocked out 'electron' module, which I think makes it a lot more straight forward. This case is a little different from [netlify.test.ts](https://github.com/stefanjudis/netlify-menubar/blob/master/src/netlify.test.ts#L8-L16) because the tests here need to keep a handle on the mock functions to make assertions about whether they were called.

Seems reasonable to me, but I haven't written all that many tests, so let me know if you think I'm doing something naive/foolish/crazy here 😆

Reply:
Looks reasonable! 👌